engine/names/src/lib.rs

```diff
@@ -1,11 +1,43 @@
 #![allow(dead_code)]
 #![feature(try_trait_v2)]
 #![feature(allocator_api)]
+
+use hax_lib_protocol::crypto::*;
 extern crate alloc;
 /* This is a dummy Rust file. Every path used in this file will be
  * exported and made available automatically in OCaml. */
 
 fn dummy_hax_concrete_ident_wrapper<I: core::iter::Iterator<Item = u8>>(x: I, mut y: I) {
+    // Crypto abstraction layer names
+    let bytes = vec![0u8; 32];
+    let iv = AEADIV::from_bytes(&bytes);
+    let key = AEADKey::from_bytes(AEADAlgorithm::Chacha20Poly1305, &bytes);
+
+    let (cipher_text, tag) = aead_encrypt(key, iv, &bytes, &bytes);
+    let iv = AEADIV::from_bytes(&bytes);
+    let key = AEADKey::from_bytes(AEADAlgorithm::Chacha20Poly1305, &bytes);
+    let _ = aead_decrypt(key, iv, &bytes, &cipher_text, AEADTag::from_bytes(&bytes));
+
+    let p = DHElement::from_bytes(&bytes);
+    let s = DHScalar::from_bytes(&bytes);
+    dh_scalar_multiply(DHGroup::X25519, s.clone(), p);
+    dh_scalar_multiply_base(DHGroup::X25519, s);
+
+    let _ = hmac(HMACAlgorithm::Sha256, &bytes, &bytes);
+
+    let _ = 1u64.to_le_bytes();
+    let slice = &bytes[0..1];
+    let _ = slice.len();
+    let _ = slice.to_vec();
+    let _ = [slice, slice].concat();
+    let mut v = vec![0];
+    v.extend_from_slice(slice);
+    v.truncate(1);
+
+    let _ = hash(HashAlgorithm::Sha256, &bytes);
+    let _ = cipher_text.clone();
+    // End crypto abstractions example
```

Comment:
You can put this code in a separate `crypto_abstractions.rs` (and add a `mod crypto_abstractions;` to that file) if you like, it would be nicer than begin-end comments IMO :)

And you could `use hax_lib_protocol::crypto::*` only there.